### DISCUSSION TITLE
two answers

### DISCUSSION THREAD

**Discussion by Bubble2** [wevm/wagmi#1661] [Q&A]
1. Why can't I get the chainId when I don't connect the wallet
2. ConfigureChains configures the specified network, and metamask switches to an unsupported network, and there is no unsupported response value

**Comment by tmm** [comment]
> 1. Why can't I get the chainId when I don't connect the wallet

wagmi only is aware of the chain id when there is a connected wallet. If you want to allow users to change the chain id without being connected, I suggest keeping track of this using some type of global store (e.g. React Context, Redux, Zustand). You can then pass that chain id to various hooks that support the `chainId` prop and sync it when there is a connected wallet using `useNetwork`.

> 2. ConfigureChains configures the specified network, and metamask switches to an unsupported network, and there is no unsupported response value

Can you drop a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example)? (Either link to CodeSandbox or public GitHub repo with instructions on how to run.)
